app/src/main/java/protect/card_locker/ScanActivity.java

```diff
@@ -221,4 +240,36 @@ public void addFromImage(View view) {
             Log.e(TAG, "No activity found to handle intent", e);
         }
     }
+
+    private void promptCameraPermission() {
+        if (camPermissionToast == null)
+            camPermissionToast = Toast.makeText(this, R.string.noCameraPermissionError, Toast.LENGTH_SHORT);
+        camPermissionToast.show();
+    }
+
+
+
+    private void changeStateBasedOnCamPermission(boolean hasGranted) {
```

Comment:
I think this function name is a bit unclear, maybe:

```suggestion
    private void showCameraPermissionMissingText(boolean show) {
```

Reply:
Since this function would not only show the camera permission missing text but also removes the camera permission missing text when the permission is granted, I think maybe calling this function a name that suggests both actions is better.

Reply:
Naming things, one of the most difficult challenges in programming :)

For me, `showCameraPermissionMissingText` with a `show` boolean explains what happens (show or hide the permission text) much better than `changeStateBasedOnCamPermission` (change which "state"? What is "state" in this context?)

Throughout Catima I have used the `showSomething(boolean show)` pattern to either show or hide a lot, so it would be more consistent with the rest of the codebase too. I am curious what other name you would suggest though.

Reply:
That's true, naming is always the biggest challenge for me in programming. 

It's always good and can cause less confusion if we keep the naming consistent. Let's just use the name you suggested.